environment.go

```diff
@@ -86,11 +86,41 @@ func (b Builder) newEnvironment(ctx context.Context) (*environment, error) {
 	// write the main module file to temporary folder
 	mainPath := filepath.Join(tempFolder, "main.go")
 	log.Printf("[INFO] Writing main module: %s\n%s", mainPath, buf.Bytes())
-	err = os.WriteFile(mainPath, buf.Bytes(), 0644)
+	err = os.WriteFile(mainPath, buf.Bytes(), 0o644)
 	if err != nil {
 		return nil, err
 	}
 
+	if len(b.EmbedDirs) > 0 {
+		for _, d := range b.EmbedDirs {
+			err = copy(d.Dir, filepath.Join(tempFolder, "files", d.Name))
+			if err != nil {
+				return nil, err
+			}
+			_, err = os.Stat(d.Dir)
+			if err != nil {
+				return nil, fmt.Errorf("embed directory does not exist: %s", d.Dir)
+			}
+			log.Printf("[INFO] Embedding directory: %s", d.Dir)
+
+			var buf bytes.Buffer
+			tpl, err := template.New("embed").Parse(embeddedModuleTemplate)
```

Comment:
Don't merge yet. I need to fix the `err` shadowing here, which prevents the deferred cleanup because it's not the same `err.